module/modmanager/manager.go

```diff
@@ -255,6 +259,9 @@ func (mgr *Manager) Add(ctx context.Context, confs ...config.Module) error {
 		// The config was already validated, but we must check again before attempting to add.
 		if err := conf.Validate(""); err != nil {
 			mgr.logger.CErrorw(ctx, "Module config validation error; skipping", "module", conf.Name, "error", err)
+			mgr.muFailedModules.Lock()
```

Comment:
maybe factor out these three lines? seems a little clunky to have to keep repeating it and any changes to the steps will require changes in so many places